packages/medusa/src/strategies/product/import.ts

```diff
@@ -0,0 +1,772 @@
+    transactionManager: EntityManager
+  ): Promise<void> {
+    const productOptionRepo = this.manager_.getCustomRepository(
+      this.productOptionRepo_
+    )
+
+    const variantOps = await this.getImportDataFromRedis(
+      batchJobId,
+      OperationType.VariantUpdate
+    )
+
+    for (const variantOp of variantOps) {
+      try {
+        await this.prepareVariantOptions(variantOp, productOptionRepo)
+
+        await this.productVariantService_
+          .withTransaction(transactionManager)
+          .update(
+            variantOp["variant.id"] as string,
+            transformVariantData(variantOp) as UpdateProductVariantInput
+          )
+      } catch (e) {
+        ProductImportStrategy.throwDescriptiveError(variantOp)
+      }
+
+      this.updateProgress_(batchJobId)
+    }
+  }
+
+  /**
+   * Extend records used for creating variant options with corresponding product option ids.
+   *
+   * @param variantOp - Parsed row data form CSV
+   * @param productOptionRepo - ProductOption repository
+   */
+  protected async prepareVariantOptions(
+    variantOp,
+    productOptionRepo: ProductOptionRepository
+  ): Promise<void> {
+    const productOptions = variantOp["variant.options"] || []
+
+    for (const o of productOptions) {
+      const { id } = (await productOptionRepo.findOne({
+        where: { title: o._title },
+      })) as ProductOption
+
+      o.option_id = id
+    }
+  }
+
+  /**
+   * Store import ops JSON to Redis.
+   * Data will expire after an hour.
+   *
+   * @param batchJobId - An id of the current batch job being processed.
+   * @param results - An object containing parsed CSV data.
+   */
+  async setImportDataToRedis(
+    batchJobId: string,
+    results: Record<OperationType, TParsedRowData[]>
+  ): Promise<void> {
+    for (const op in results) {
+      if (results[op]?.length) {
+        await this.redisClient_.set(
+          `pij_${batchJobId}:${op}`,
+          JSON.stringify(results[op]),
+          "EX",
+          60 * 60
+        )
+      }
+    }
+  }
+
+  /**
+   * Retrieve parsed CSV data from Redis.
+   *
+   * @param batchJobId - An id of the current batch job being processed.
+   * @param op - Type of import operation.
+   */
+  async getImportDataFromRedis(
+    batchJobId: string,
+    op: OperationType
+  ): Promise<TParsedRowData[]> {
+    return JSON.parse(
+      (await this.redisClient_.get(`pij_${batchJobId}:${op}`)) || "[]"
+    )
+  }
+
+  /**
+   * Clear all batch job data set by the strategy.
+   *
+   * @param batchJobId - An id of the current batch job being processed.
+   */
+  async clearRedisRecords(batchJobId: string): Promise<number> {
+    return await this.redisClient_.del(`pij_${batchJobId}:*`)
+  }
+
+  /**
+   * Update count of processed data in the batch job context
+   * and cleanup Redis data.
+   *
+   * @param batchJobId - An id of the current batch job being processed.
+   */
+  private async finalize_(batchJobId: string): Promise<void> {
```

Comment:
todo: As mention in the convention, we don't need to suffix the non public method of a class with `_` :)